crates/analyzer-host/src/fsm.rs

```diff
@@ -104,3 +104,66 @@ impl LspProtocolMachine {
 		)
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use json_rpc::message::Request;
+	use json_rpc::message::Notification;
+	use serde_json::Value;
+	
+	use crate::{fs::LspEnumerableFileSystem, json_rpc};
+	use super::*;
+	#[test]
+	fn test_is_active() {
+		let rm = RequestManager::new(async_channel::unbounded::<Message>());
+		let mut lsp = LspProtocolMachine::new(None, rm.clone(),
+													Arc::new(Box::new(LspEnumerableFileSystem::new(rm.clone()))));
+		assert_eq!(lsp.is_active(), true);
+
+		lsp.current_state = LspServerState::Stopped;
+		assert_eq!(lsp.is_active(), false);
```

Comment:
Could you send a request to the server here instead? `is_active()` is literally `self.current_state != Stopped` 😄